rain_api_core/auth.py

```diff
@@ -0,0 +1,154 @@
+        except jwt.InvalidSignatureError:
+            log.info('JWT has failed verification')
+        return None
+
+    def _encode_jwt(self, payload: Mapping[str, str]):
+        try:
+            encoded = jwt.encode(payload, self.private_key, self.algorithm)
+        except TypeError:
+            log.error('unable to encode jwt cookie')
+            return ''
+        return encoded
+
+    @staticmethod
+    def _format_date_for_cookie_header(timestamp):
+        return format_7231_date(timestamp)
+
+    def _jwt_payload_from_user_profile(self, user_profile: Union[UserProfile, None]):
+        if user_profile is None:
+            return {}
+        now = int(time())
+        return {
+            'urs-user-id': user_profile.user_id,
+            'first_name': user_profile.first_name,
+            'last_name': user_profile.last_name,
+            'email': user_profile.email,
+            'urs-access-token': user_profile.token,
+            'urs-groups': user_profile.groups,
+            'iat': now,
+            'exp': now + self.session_ttl
+        }
+
+    def _retrieve_blacklist(self):
+        if self.black_list and self.black_list_last_retrieved <= time() + (10 * 60):  # cached in last 10 minutes
+            pass
+        response = urllib.request.urlopen(self.black_list_endpoint).read().decode('utf-8')  # nosec
+        self.black_list = json.loads(response)['blacklist']
+        self.black_list_last_retrieved = time()
+
+    def _in_blacklist(self, user_profile: UserProfile):
+        self._retrieve_blacklist()
+        user_blacklist_time = self.black_list.get(user_profile.user_id)
+        if user_blacklist_time is not None:
+            if user_blacklist_time >= user_profile.iat:
+                return True
+        return False
+
+    def get_profile_from_headers(self, headers) -> Union[UserProfile, None]:
+        """Inspects headers for auth cookie and return user_profile if authenticated, None otherwise"""
+        auth_cookie = self._get_auth_cookie(headers)
+        if not auth_cookie:
+            return None
+
+        jwt_decoded = self._decode_jwt(auth_cookie.value)
+
+        if jwt_decoded is None:
+            return None
+        user_profile = UserProfile.from_jwt_payload(jwt_decoded)
+        if self._in_blacklist(user_profile):
+            return None
+        return user_profile
+
+    def get_header_to_set_auth_cookie(self, user_profile: Union[UserProfile, None], cookie_domain=''):
+        """ Gets a header to set auth-cookie
+
+        Parameters:
+        UserProfile: UserProfile to use in construction of a cookie, if none will return header to unset/logout
+        """
+        payload = self._jwt_payload_from_user_profile(user_profile)
+        cookie_value = self._encode_jwt(payload) if payload else 'expired'
+        cookie_domain = f'; Domain={cookie_domain}' if cookie_domain else ''
+        if payload:
+            expire_date = JwtManager._format_date_for_cookie_header(payload['exp'])
```

Comment:
I already suggested scrapping the `staticmethod` all together, but since it's a static method you might as well call it like this:
```suggestion
            expire_date = self._format_date_for_cookie_header(payload['exp'])
```